File: lib/biz/spec-iot/src/main/java/site/addzero/biz/spec/iot/tdengine/SqlStatement.java
package site.addzero.biz.spec.iot.tdengine;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

/**
 * SQL string plus ordered parameter list.
 */
public final class SqlStatement {

    private final String sql;
    private final List<Object> parameters;

    public SqlStatement(String sql, List<Object> parameters) {
        if (sql == null || sql.trim().isEmpty()) {
            throw new IllegalArgumentException("sql must not be blank");
        }
        this.sql = sql;
        this.parameters = Collections.unmodifiableList(new ArrayList<Object>(parameters));
    }

    public String getSql() {
        return sql;
    }

    public List<Object> getParameters() {
        return parameters;
    }
}

File: lib/biz/spec-iot/src/main/kotlin/site/addzero/biz/spec/iot/tdengine/SqlStatement.kt
package site.addzero.biz.spec.iot.tdengine

import site.addzero.biz.spec.iot.requireText

/**
 * SQL string plus ordered parameter list.
 */
class SqlStatement(
    sql: String?,
    parameters: List<Any?>,
) {

    val sql: String = requireText(sql, "sql")
    val parameters: List<Any?> = ArrayList(parameters).toList()
}
